### PR TITLE
Update BrokerHost.java to work with the new BrokerHost.

### PR DESCRIPTION
**Pull request by p3dr0rv**
In order to test the MWJ API we modify the broker Host (see https://github.com/AzureAD/ad-accounts-for-android/pull/2111),
These changes affected the UI of the testing app, so a rework in the brokerHost.java was needed.

Related PR: https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/1803

tested changes here:
https://identitydivision.visualstudio.com/Engineering/_build/results?buildId=1084514&view=results
Note: the failing test are not related to BrokerHost.java

### CODE REVIEW

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerHost.java [hunk 1]
```diff
@@ -33,56 +33,51 @@
...
     public final static String BROKER_HOST_APK = "BrokerHost.apk";
     public final static String OLD_BROKER_HOST_APK = "OldBrokerHost.apk";
     public final static String BROKER_HOST_APK_PROD = "BrokerHostProd.apk";
     public final static String BROKER_HOST_APK_RC = "BrokerHostRC.apk";
+    // Fragments for BrokerHost
+    public final BrokerFlightsFragment brokerFlightsFragment;
+    public final BrokerApiFragment brokerApiFragment;
+    public final SingleWpjApiFragment singleWpjApiFragment;
 
     public BrokerHost() {
```
Comment (fadidurah) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerHost.java:60]: I think this class should remain as the only class devs need to interact with to write automation for UI tests using BrokerHost. Maybe we can make the new classes package scope

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/AbstractBrokerHost.java [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+        UiAutomatorUtils.handleButtonClick(resourceId);
+    }
+
+    /**
+     * This method fills the text box with the given resource id with the given text
+     *
+     * @param resourceEditText the resource id of the text box
+     * @param text             the text to be write in the text box
+     */
+    public void fillTextBox(@NonNull final String resourceEditText, @NonNull final String text) {
```
Comment (fadidurah) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/AbstractBrokerHost.java]: Static?

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/AbstractBrokerHost.java [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+        @Getter
+        private final String title;
+    }
+
+    /**
+     * This method clicks on the button with the given resource id
+     *
+     * @param resourceIdButton the resource id of the button to be clicked
+     */
+    public void clickButton(@NonNull final String resourceIdButton) {
```
Comment (fadidurah) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/AbstractBrokerHost.java]: Static?

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/AbstractBrokerHost.java (hunk 2 above)
Comment (fadidurah) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/AbstractBrokerHost.java]: Consider this for all of these methods handling UI

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerHost.java [hunk 4]
```diff
@@ -94,183 +89,39 @@ public void performDeviceRegistration(@NonNull final String username,
...
-        final UiObject certInstaller = device.findObject(new UiSelector().packageName("com.android.certinstaller"));
-        certInstaller.waitForExists(FIND_UI_ELEMENT_TIMEOUT);
-        Assert.assertTrue(
-                "Microsoft Authenticator - cert installer dialog appears.",
-                certInstaller.exists()
-        );
-
-        UiAutomatorUtils.handleButtonClick("android:id/button1");
+        singleWpjApiFragment.launch();
+        singleWpjApiFragment.installCertificate();
```
Comment (fadidurah) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerHost.java:124]: This naming seems inconsistent, even in the old code. Enable browser access is the same as installing cert?

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/SingleWpjApiFragment.java [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+        clickButton(GET_DEVICE_STATE_BUTTON_ID);
+        return dismissDialogBoxAndGetText();
+    }
+
+    /**
+     * Checks if the device is shared.
+     *
+     * @return true if the device is shared, false otherwise.
+     */
+    public boolean isDeviceShared() {
```
Comment (fadidurah) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/SingleWpjApiFragment.java:97]: Do we have a way to access this from BrokerHost.java? Let's keep that class as the only one that devs need to access

**Review by fadidurah — APPROVED** [review]
Some comments

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/AbstractBrokerHost.java [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+            return dialogBoxSplit.length > 1 ? dialogBoxSplit[1] : dialogBoxText;
+        } catch (final UiObjectNotFoundException e) {
+            throw new AssertionError(e);
+        } finally {
+            // dismiss dialog
+            UiAutomatorUtils.handleButtonClick(DIALOG_BOX_OK_BUTTON_RESOURCE_ID);
+        }
+    }
+
+    static public void delay(final long delayInSeconds) {
```
Comment (iamgusain) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/AbstractBrokerHost.java]: There is a method in ThreadUtils.java 'sleepSafely', we can use that.

**Review by iamgusain — APPROVED** [review]

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/AbstractBrokerHost.java (hunk 6 above)
Comment (p3dr0rv) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/AbstractBrokerHost.java]: done

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/AbstractBrokerHost.java (hunk 3 above)
Comment (p3dr0rv) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/AbstractBrokerHost.java]: done

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/AbstractBrokerHost.java (hunk 2 above)
Comment (p3dr0rv) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/AbstractBrokerHost.java]: done

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/SingleWpjApiFragment.java (hunk 5 above)
Comment (p3dr0rv) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/SingleWpjApiFragment.java:97]: BrokerHost.java remains as the entry point for all operation related to the broker

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerHost.java (hunk 1 above)
Comment (p3dr0rv) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerHost.java:60]: BrokerHost.java remains as the entry point for all operation related to the broker.

**Pull request merged**